cscs-checks/system/basic/file_system_check.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+import getpass
+import os
+import re
+
+import reframe as rfm
+import reframe.utility.osext as osext
+import reframe.utility.sanity as sn
+
+
+class FileSystemGlobal(rfm.RegressionMixin):
+    '''Handy class to store common test settings.
+    '''
+    scratch = parameter(variable(list, value=['SCRATCH']))
+
+
+class FileSystemCommandCheck(rfm.RunOnlyRegressionTest):
+    # TODO: test from cn as well
+    valid_systems = ['daint:login', 'dom:login']
+    valid_prog_environs = ['builtin']
+    executable = 'time -p'
+    tags = {'ops', 'diagnostic', 'health'}
+    maintainers = ['CB', 'VH']
+
+    @rfm.run_before('sanity')
+    def set_sanity_and_perf(self):
+        self.sanity_patterns = sn.assert_eq(self.job.exitcode, 0)
+        self.perf_patterns = {
+            'real_time': sn.extractsingle(r'real (?P<real_time>\S+)',
+                                          self.stderr, 'real_time', float)
+        }
+
+
+@rfm.simple_test
+class fs_check_cd_dir(FileSystemCommandCheck, FileSystemGlobal):
+    reference = {
+        'daint:login': {
+            'real_time': (0.1, None, 0.1, 's')
+        },
+        'dom:login': {
+            'real_time': (0.1, None, 0.1, 's')
+        }
+    }
+
+    @rfm.run_before('run')
+    def set_executable_ops(self):
+        self.executable_opts = ['cd', osext.expandvars(f'${self.scratch}')]
+
+
+@rfm.simple_test
+class fs_check_ls_dir(FileSystemCommandCheck, FileSystemGlobal):
+    reference = {
+        'daint:login': {
+            'real_time': (0.1, None, 0.1, 's')
+        },
+        'dom:login': {
+            'real_time': (0.1, None, 0.1, 's')
+        }
+    }
+
+    @rfm.run_before('run')
+    def set_executable_ops(self):
+        self.executable_opts = ['/usr/bin/ls',
+                                osext.expandvars(f'${self.scratch}')]
+
+
+@rfm.simple_test
+class fs_check_du_dir(FileSystemCommandCheck, FileSystemGlobal):
+    directory = parameter(['PROJECT',
+                           'HOME',
+                           'SCRATCH'])
+    # TODO: system is not always relevant
+    reference = {
```

Comment:
Name this as `fs_reference` and make it a variable. Then you set the `reference` in a hook.

Reply:
Done

Reply:
I'm not sure I understand this change: the reference in the hook, used by reframe for the performance check must be a dict if I'm not wrong (I get a python error if the type is not like that). For example, I can make it work by defining reference as 

self.reference = {self.directory: self.fs_reference[self.directory]}

where self.directory is a parameter and where self.fs_reference is the new variable. Is that correct? Thanks!